helios-integrations/helios-retrofit/src/main/kotlin/helios/retrofit/HeliosResponseBodyConverter.kt

```diff
@@ -1,24 +1,21 @@
 package helios.retrofit
 
 import arrow.core.flatMap
-import arrow.core.getOrElse
 import helios.core.Json
 import helios.typeclasses.Decoder
 import okhttp3.ResponseBody
 import retrofit2.Converter
+import java.io.IOException
 import java.nio.ByteBuffer
 
 class HeliosResponseBodyConverter<T>(private val decoder: Decoder<T>) : Converter<ResponseBody, T> {
 
   // TODO: Refactor this block when Arrow's Resource data type is available
-  override fun convert(value: ResponseBody): T? = try {
+  override fun convert(value: ResponseBody): T = value.use { body ->
     Json
-      .parseFromByteBuffer(ByteBuffer.wrap(value.byteStream().readBytes()))
-      .flatMap { it.decode(decoder) }
-      .getOrElse { null }
-  } catch (ex: Exception) {
-    null
-  } finally {
-    value.close()
+      .parseFromByteBuffer(ByteBuffer.wrap(body.byteStream().readBytes()))
+      .flatMap { json -> json.decode(decoder).mapLeft { IOException("Decode failed with the error: $it") } }
+      .fold({ throw IOException(it) }, { it })
```

Comment:
This is creating an `IOException` inside of an `IOException`. Should this be something like:

```suggestion
      .fold({ throw it }, { it })
```

Do we have a `rethrow` like in cats?

Reply:
@fedefernandez We do in the SNAPSHOT Arrow but not in stable.

Reply:
I'll add a TODO